### PR TITLE
chore(es): allow optional use of Amazon open distro images

### PR DESCRIPTION
**Pull request by volkflo**
This should allow to switch between Amazon OpenDistro and the official elastic images for the test runs

### CODE REVIEW

File: test/common/pom.xml [hunk 1]
```diff
@@ -97,5 +101,23 @@
...
+      <artifactId>testcontainers</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>elasticsearch</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.testcontainers</groupId>
+      <artifactId>elasticsearch</artifactId>
+      <scope>compile</scope>
```
Comment (msavy) [inline test/common/pom.xml:115]: I'm still not sure about including these deps twice in this way. It's very odd 😅. I will try to figure out whether it's possible to leave out one of the two.

**Review by msavy — APPROVED** [review]

**Comment by msavy** [comment]
Thanks for this 🚀.

**Comment by msavy** [comment]
I made a PR against your branch which should fix the NPE issue.

File: test/common/pom.xml (hunk 1 above)
Comment (volkflo) [reply test/common/pom.xml:115]: Yes, this is very strange, but I was not able to build it without this.

File: common/es/src/main/java/io/apiman/common/es/util/EsConstants.java [hunk 2]
```diff
@@ -244,30 +245,24 @@
...
-            java.util.Properties p = new Properties();
-            String version;
+    public static Properties getEsVersion() {
+        if (esVersions == null) {
+            InputStream is = EsConstants.class.getResourceAsStream("apiman-es.properties");
+            Properties p = new Properties();
             try {
                 p.load(is);
             } catch (IOException e) {
+                e.printStackTrace();
```
Comment (msavy) [inline common/es/src/main/java/io/apiman/common/es/util/EsConstants.java:262]: If this is an unrecoverable error, then we should wrap the `IOException` in `UncheckedUIException` and rethrow it.

If it is ignored, then I suggest you either log it at trace level or do nothing.

**Review by msavy — CHANGES_REQUESTED** [review]

File: common/es/src/main/java/io/apiman/common/es/util/EsConstants.java (hunk 2 above)
Comment (volkflo) [reply common/es/src/main/java/io/apiman/common/es/util/EsConstants.java:262]: I am not totally sure if I understand you correctly. So this method is only used for tests, I think a `UncheckedUIException` is even more confusing?

Comment (volkflo) [reply common/es/src/main/java/io/apiman/common/es/util/EsConstants.java]: Changed it :)

**Review by msavy — APPROVED** [review]
Thanks, this is good to go 👍🚀

**Pull request merged**